images/linux/scripts/SoftwareReport/SoftwareReport.Common.psm1

```diff
@@ -331,6 +331,22 @@ function Get-DotNetCoreSdkVersions {
     return $dotNetCoreSdkVersion
 }
 
+function Get-DotnetTools {
+
+    $dotnetToolset = (Get-ToolsetContent).dotnet
+    $dotnetTools = $dotnetToolset.tools
+
+    $toolsList = @()
+
+    ForEach ($dotnetTool in $dotnetTools)
+    {
+        (dotnet tool update -g $dotnetTool | Out-String) -match "was reinstalled with the latest stable version \(version \'(?<version>\d+\.\d+\.\d+)" | Out-Null
```

Comment:
We shouldn't create an unpredictable situation and update the packages.

Reply:
This updates the tool specifically. Anything else. Even (in the very unlikely event!) that a new version of the tool was been release since the install that happened a few moments ago, there's no harm done. The dotnet tool will have the latest version installed.

Anyway, the fundamental reason for this being here it's because there is no other way of capturing the version of a dotnet tool. Unless one captures it when installing it.

Reply:
It's definitely unpredictable and difficult to debug, because the `RunAll-Tests.ps1` script invokes before `SoftwareReport.Generator.ps1` script.

Reply:
OK... but the tool has been installed before `RunAll-Tests.ps1` and if the install has failed for some reason, the test will fail, therefore it won't reach this.

If you want to grab the tool version at install time, the only way I can think of persisting the version is to add to a env var and then use it here to output it.
Or just let me know what's your proposal to fix this. 😅

Reply:
Why can't we call `nbgv` and get version?

```
PS > nbgv --version
3.4.244+c60a07221c
```

Reply:
That will work for nbgv, sure. But what to do when you add other dotnet tools? We would need a test matrix on what to throw to each tool and what to expect back... `--version` it's implemented by nbgv it's not because it's a dotnet tool.

Reply:
Yep, that's right we should create a basic test and validate the expected output result for each tools.

Reply:
Specific command to get version from each dotnet tool implemented. 
Please note that this will rely on each dotnet tools used to provide a way to output a "clean" version string, otherwise it will have to handled differently.